subsys/shell/Kconfig

```diff
@@ -56,6 +56,24 @@ config SHELL_BACKSPACE_MODE_DELETE
 	  Some terminals send code: 0x08 (backspace) other 0x7F (delete). When
 	  this option is set shell will expect 0x7F for backspace key.
 
+config SHELL_PROMPT_CHANGE
+	bool "Allow prompt change in runtime"
+	default y if !SHELL_MINIMAL
```

Comment:
I have enabled this option by default because, all things considered, it's a bug fix. However, I won't insist on it since the probability of its occurrence is very low.